### PR TITLE
wrapper for nodes

### PR DESCRIPTION
**Pull request by scuben**
I was in need to have a wrapper div on the node level containing all content elements within the node. This way I am able to define id/class to the node for styling. Then I can include this node anywhere and have it styled without any further configuration (like doing the wrapper when including the node). I hope you get the idea behind :)

### CODE REVIEW

**Comment by Toflar** [comment]
Haven't we introduced wrappers already in https://github.com/terminal42/contao-node/issues/13?
What's different here? :)

**Comment by scuben** [comment]
The implementation in #13 does the same but you need to do that everywhere you include nodes. This is not practicable when including a node on several pages where you need to have a wrapper with a consistent class. This PR implements a wrapper configuration on the node level to encapsulate the content elements within the node definition.

**Comment by Toflar** [comment]
I see. Implementation-wise I would also like to have that template based, so that's one thing to adjust for sure.
But conceptual-wise this sounds okay to me. Wdyt @qzminski?

**Comment by scuben** [comment]
I see your point. But I might need a little kickstart info to how I should implement a template based info.

**Comment by qzminski** [comment]
I would also like to see an extra field in the `wrapper` subpalette where you can select a partial frontend template (e.g. `node_default.html5`) used to render that wrapper. So if `wrapper` is enabled, the `$buffer` in `generateBuffer()` method should become:

```php
if ($nodeModel->wrapper) {
    $template = new FrontendTemplate($nodeModel->template);
    $template->setData($nodeModel->row());
    $template->buffer = $buffer;

    $buffer = $template->parse(); // to be returned
}
```

**Comment by qzminski** [comment]
Look pretty okay to me, @Toflar ?

File: src/NodeManager.php [hunk 1]
```diff
@@ -80,6 +82,7 @@ public function generateMultiple(array $ids): array
      */
     private function generateBuffer(NodeModel $nodeModel): string
     {
+        $nodeTpl = 'node_default';
```
Comment (Toflar) [inline src/NodeManager.php:85]: I guess we don't need this variable, do we? We can just use it on line 99 directly :)

File: src/Resources/contao/dca/tl_node.php [hunk 2]
```diff
@@ -145,6 +151,29 @@
...
+            'sql' => ['type' => 'string', 'length' => 1, 'default' => ''],
+        ],
+        'nodeTpl' => [
+            'exclude' => true,
+            'inputType' => 'select',
+            'options_callback' => static function () {
+                return Contao\Controller::getTemplateGroup('node_');
+            },
+            'eval' => ['includeBlankOption' => true, 'chosen' => true, 'tl_class' => 'w50'],
+            'sql' => ['type' => 'string', 'length' => 64, 'defualt' => ''],
```
Comment (Toflar) [inline src/Resources/contao/dca/tl_node.php:168]: Typo `default`

File: src/Resources/contao/dca/tl_node.php [hunk 3]
```diff
@@ -145,6 +151,29 @@
...
+        ],
+        'cssID' => [
+            'label' => &$GLOBALS['TL_LANG']['tl_node']['cssID'],
+            'exclude' => true,
+            'inputType' => 'text',
+            'eval' => ['multiple' => true, 'size' => 2, 'tl_class' => 'w50'],
+            'sql' => ['type' => 'string', 'length' => 255, 'defualt' => ''],
```
Comment (Toflar) [inline src/Resources/contao/dca/tl_node.php:175]: Typo `default`

**Comment by qzminski** [comment]
1.2.0 released, thank you @scuben !

**Pull request merged**